File: Products/math-thinking-coach/backend/app/schemas/topic.py
from pydantic import BaseModel, Field


class LearningObjective(BaseModel):
    id: str
    text: str


class Concept(BaseModel):
    id: str
    title: str
    body: str
    learningObjectives: list[LearningObjective]


class WorkedExample(BaseModel):
    id: str
    conceptId: str
    problem: str
    steps: list[str]
    finalAnswer: str


class Topic(BaseModel):
    id: str
    chapterId: str
    title: str
    explanation: str
    workedExampleContent: str
    learningObjectives: list[str]
    # Additive, Slice A1 (Structured Learning Content Foundation): structured
    # equivalents of the three legacy fields above. Default to an empty list
    # so existing runtime topics.json entries for not-yet-migrated chapters
    # (only squares-and-cubes is migrated in A1) keep validating unchanged.
    # explanation/workedExampleContent/learningObjectives are removed only in
    # Slice A3, once every Topic-bearing chapter has migrated - see
    # docs/Structured-Learning-Content-Design-Proposal.md §K/§M.
    concepts: list[Concept] = Field(default_factory=list)
    workedExamples: list[WorkedExample] = Field(default_factory=list)

File: Products/math-thinking-coach/frontend/src/types/topic.ts
export interface LearningObjective {
  id: string;
  text: string;
}

export interface Concept {
  id: string;
  title: string;
  body: string;
  learningObjectives: LearningObjective[];
}

export interface WorkedExample {
  id: string;
  conceptId: string;
  problem: string;
  steps: string[];
  finalAnswer: string;
}

export interface Topic {
  id: string;
  chapterId: string;
  title: string;
  explanation: string;
  workedExampleContent: string;
  learningObjectives: string[];
  // Additive, Slice A1: structured equivalents of the three legacy fields
  // above. Always present on the API response (backend defaults to []), but
  // only actually populated for chapters that have migrated (Slice A1
  // pilot: A Square and A Cube) - empty for every other Topic-bearing
  // chapter until its own migration (Slice A2b, not yet authorized). See
  // docs/Structured-Learning-Content-Design-Proposal.md §K/§M/§W.
  concepts: Concept[];
  workedExamples: WorkedExample[];
}
